internal/statistics/metrics.go

```diff
@@ -43,24 +79,46 @@ func init() {
 	prometheus.Unregister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
 	prometheus.Unregister(collectors.NewGoCollector())
 
-	prometheus.MustRegister(WalgMetrics.uploadedFilesTotal)
-	prometheus.MustRegister(WalgMetrics.uploadedFilesFailedTotal)
+	prometheus.MustRegister(WalgMetrics.UploadedFilesTotal)
+	prometheus.MustRegister(WalgMetrics.UploadedFilesFailedTotal)
+
+	prometheus.MustRegister(WalgMetrics.S3Code200)
+	prometheus.MustRegister(WalgMetrics.S3Code400)
+	prometheus.MustRegister(WalgMetrics.S3Code404)
+	prometheus.MustRegister(WalgMetrics.S3Code429)
+	prometheus.MustRegister(WalgMetrics.S3Code500)
 }
 
 func PushMetrics() {
-	address := viper.GetString(StatsdAddressSetting)
+	address := viper.GetString(conf.StatsdAddressSetting)
 	if address == "" {
 		return
 	}
 
-	extraTags := viper.GetStringMapString(StatsdExtraTagsSetting)
+	extraTags := viper.GetStringMapString(conf.StatsdExtraTagsSetting)
 
 	err := pushMetrics(address, extraTags)
 	if err != nil {
 		tracelog.WarningLogger.Printf("Pushing metrics failed: %v", err)
 	}
 }
 
+func WriteStatusCodeMetric(code int) {
+	if code >= 500 {
+		WalgMetrics.S3Code500.Inc()
+	} else if code >= 400 {
+		if code == 404 {
+			WalgMetrics.S3Code404.Inc()
+		} else if code == 429 {
+			WalgMetrics.S3Code429.Inc()
+		} else {
+			WalgMetrics.S3Code400.Inc()
```

Comment:
It's a bit confusing to collapse all 4xx codes (but 429) to one counter.

May be it's better to use `prometheus.GaugeVec`  https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#GaugeVec
for fair http code statistics ?

```
WalgMetrics.S3Codes.WithLabelValues(strconv.Itoa(code)).Inc()
```